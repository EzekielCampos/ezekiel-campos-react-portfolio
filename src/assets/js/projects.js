import blog from "../img/tech-blog-main.png";
import listIt from "../img/list-it-home-pg.png";
import bonAppetit from "../img/bon-appetit-page.png";

export const projectData = [
  {
    id: 1,
    name: "Social Network API Using Mongoose",
    description:
      "This project focuses on creating an API for a social networking web application. The backend was built with MongoDB in association with Mongoose for creating models and organizing data.This assignment was meant to create a functioning backend using Mongoose where users can create a friend’s list, and sharing and reacting to other user’s thoughts.",
    video:
      "https://drive.google.com/file/d/1sNV5Q-qAhhKLSudDqTvVZynQIrGg76K6/preview",
    deployed:
      "https://drive.google.com/file/d/1sNV5Q-qAhhKLSudDqTvVZynQIrGg76K6/preview",
    github: "https://github.com/EzekielCampos/social-network-mongo-api",
  },
  {
    id: 2,
    name: "List It",
    description:
      " This full-stack application allows users to create their account so that they can keep track of all their nerdy hobbies.The purpose of this project was to create an application using the Model-View-Controller(MVC) concept that seamlessly connects the front and back end intuitively. ",
    pic: listIt,
    github: "https://github.com/EzekielCampos/List-It",
    deployed: "https://list-it-kw5e.onrender.com/",
  },
  {
    id: 3,
    name: "Tech Blog Application",
    description:
      "This application is a blog page where developers can share views and thoughts with other registered users of the app. It gives the option to create an account so that a user has the ability to add, update, delete their post. In addition each user can comment on other developers post. The purpose behind this project was to build a full stack application using the Model-View-Controller (MVC) paradigm. ",
    pic: blog,
    github: "https://github.com/EzekielCampos/mvc-tech-blog-app",
    deployed: "https://mvc-tech-blog-app.onrender.com/",
  },
  {
    id: 4,
    name: "E-Commerce Backend ORM",
    description:
      "This project works on creating a functioning backend for an e-commerce site that connects to a database to perform CRUD operations. The data that is being used is information about different categories, products, and tags that belong to products. The purpose behind this was to practice building a server with Express and to interface with the database using Sequelize.",
    video:
      "https://drive.google.com/file/d/1AQgrPRA8GkEkONhIwqgE5h5lphJOMZEr/preview",
    github:
      "https://github.com/EzekielCampos/ecommerce-backend-application-ORM",
    deployed:
      "https://drive.google.com/file/d/1AQgrPRA8GkEkONhIwqgE5h5lphJOMZEr/preview",
  },
  {
    id: 5,
    name: "Employee Content Management Application",
    description:
      "This project is a content management application to perform various operations pertaining to an organization employees and department. Using CLI prompts, the user has a variety of options to either read, add, or update data. The data will be held in a database and the user will have only have access to the companies information. For this assignment the main objective was to give the user access to data using PostgreSQL to perform CRUD operations. ",
    video:
      "https://drive.google.com/file/d/17czMDC6Y8p-XlYsjVGgyMJtvbwSAWAG0/preview",
    deployed:
      "https://drive.google.com/file/d/17czMDC6Y8p-XlYsjVGgyMJtvbwSAWAG0/preview",
    github:
      "https://github.com/EzekielCampos/employee-content-management-application",
  },
  {
    id: 6,
    name: "Bon Appetit",
    description:
      "This application is a food and cocktail search engine where the user can browse through various categories to find a meal or drink that might interest them.n order to build this application our team incorporated two API’s ,TheMealDB and TheCocktailDB, to fetch the information about food and drinks recipes that will be displayed on the page.",
    pic: bonAppetit,
    github:
      "https://github.com/EzekielCampos/bon-appetit-food-cocktail-recipe-search-application?tab=readme-ov-file",
    deployed:
      "https://ezekielcampos.github.io/bon-appetit-food-cocktail-recipe-search-application/",
  },
  
];
