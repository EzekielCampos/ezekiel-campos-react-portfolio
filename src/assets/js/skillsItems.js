// This data is the skills that will be displayed on the Resume page
export const skills = [
  {
    id: 1,
    tech: "JavaScript",
  },
  {
    id: 2,

    tech: "Node.js",
  },
  {
    id: 3,

    tech: "React",
  },
  {
    id: 4,

    tech: "SQL",
  },

  {
    id: 5,

    tech: "NOSQL",
  },
  {
    id: 6,

    tech: "MongoDB",
  },
  {
    id: 6,

    tech: "PostgresSQL",
  },
];

// This data is the libraries I have experience with that will be displayed on the Resume page

export const libraries = [
  {
    id: 1,
    tech: "Mongoose",
  },
  {
    id: 2,
    tech: "Material UI",
  },
  {
    id: 3,
    tech: "Express.js",
  },
  {
    id: 4,
    tech: "Vite",
  },
  {
    id: 5,
    tech: "Boostrap",
  },
  {
    id: 6,
    tech: "Sequelize",
  },
  {
    id: 7,

    tech: "Express-Session",
  },
];
