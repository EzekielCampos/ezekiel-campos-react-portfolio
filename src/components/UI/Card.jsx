import {
  Grid2 as Grid,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Typography,
  IconButton,
} from "@mui/material";
import GitHubIcon from "@mui/icons-material/GitHub";
import LanguageIcon from "@mui/icons-material/Language"; // For deployed page icon
import PropTypes from "prop-types";

export const ProjectCard = ({ data }) => {
  console.log(data);

  return data.map((data) => {
    return (
      <Grid size={{ xs: 12, md: 6, s: 4 }} key={data.id}>
        <Card sx={{ display: "flex", flexDirection: "column", height: "100%" }}>
          <CardMedia
            sx={{ height: 400, objectFit: "cover" }}
            image={data.pic}
            title={data.name}
          />

          {/* Card Content */}
          <CardContent>
            <Grid container justifyContent={"center"}>
              <Typography gutterBottom variant="h5" component="div">
                {data.name}
              </Typography>
            </Grid>
            <Grid container justifyContent={"center"}>
              <Typography variant="body2" sx={{ color: "text.secondary" }}>
                {data.description}
              </Typography>
            </Grid>
          </CardContent>

          {/* Card Actions */}
          <CardActions sx={{ justifyContent: "center", marginTop: "auto" }}>
            {/* GitHub Icon */}
            <IconButton
              sx={{ color: "skyBlue" }}
              onClick={() => window.open(data.github)}
            >
              <GitHubIcon style={{ width: 35, height: 35 }} />
            </IconButton>

            {/* Deployed Page Icon */}
            <IconButton
              sx={{ color: "#000" }}
              //   onClick={() => window.open(deployedUrl, "_blank")}
            >
              <LanguageIcon style={{ width: 35, height: 35 }} />
            </IconButton>
          </CardActions>
        </Card>
      </Grid>
    );
  });
};

ProjectCard.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string, // `title` should be a string
      description: PropTypes.string, // `description` should be a string
      github: PropTypes.string,
      pic: PropTypes.string,
    })
  ).isRequired,
};
