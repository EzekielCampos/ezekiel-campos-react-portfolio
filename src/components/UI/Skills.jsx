import {
    ListItem,
    ListItemIcon,
    ListItemText,
  } from "@mui/material";
  


  export const Skills = ({data})=>{

    return(data.map((item)=>{
        return(
            <ListItem key={item.id}>
            <ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
              •
            </ListItemIcon>
            <ListItemText>
              {item.tech}
            </ListItemText>
          </ListItem>

        )
    })
       
    )

  }



{/* <ListItem>
<ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
  •
</ListItemIcon>
<ListItemText>
  First Item
  <GitHubIcon
    style={{ width: 20, height: 20, paddingLeft: "5px" }}
  />
</ListItemText>
</ListItem>
<ListItem>
<ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
  •
</ListItemIcon>
<ListItemText fontSize={10}>
  second Item
  {/* <ListItemIcon sx={{ width: 20, height: 20, color:"black" }}>{reactIcon}</ListItemIcon> */}
  {/* React icon */}
// </ListItemText>
// </ListItem>
// <ListItem>
// <ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
//   •
// </ListItemIcon>
// <ListItemText>
//   second Item
//   <JavascriptIcon sx={{ width: 20, height: 20 }} />{" "}
// </ListItemText>
// </ListItem>

// <ListItem>
// <ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
//   •
// </ListItemIcon>
// <ListItemText>
//   second Item
//   <ListItemIcon>{mongoIcon}</ListItemIcon>
// </ListItemText>
// </ListItem> */}