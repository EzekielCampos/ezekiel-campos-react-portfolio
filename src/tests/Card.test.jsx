import { testData } from "../assets/js/projects";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import pretty from "pretty";
import { ProjectCard } from "../components/UI/Card";

describe("Project Card", () => {
  it("renders a card that will hold the information about a project", () => {
    render(<ProjectCard data={testData} />);
    const html = pretty(document.querySelector(".card").innerHTML);

    expect(html).toMatchSnapshot();
  });
});
