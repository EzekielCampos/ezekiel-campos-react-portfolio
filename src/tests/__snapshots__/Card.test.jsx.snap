// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Project Card > renders a card that will hold the information about a project 1`] = `
"<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-16wqpcm-MuiPaper-root-MuiCard-root" style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);">
  <div class="MuiCardMedia-root css-xk7vuk-MuiCardMedia-root"><iframe src="https://drive.google.com/file/d/1sNV5Q-qAhhKLSudDqTvVZynQIrGg76K6/preview" width="100%" height="100%" allow="autoplay"></iframe></div>
  <div class="MuiCardContent-root css-46bh2p-MuiCardContent-root">
    <div class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1rr1yyo-MuiGrid2-root">
      <div class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-13y79c5-MuiTypography-root">Social Network API Using Mongoose</div>
    </div>
    <div class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1rr1yyo-MuiGrid2-root">
      <p class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root">This project focuses on creating an API for a social networking web application. The backend was built with MongoDB in association with Mongoose for creating models and organizing data.This assignment was meant to create a functioning backend using Mongoose where users can create a friend’s list, and sharing and reacting to other user’s thoughts.</p>
    </div>
  </div>
  <div class="MuiCardActions-root MuiCardActions-spacing css-157ewyx-MuiCardActions-root"><button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-7f8h13-MuiButtonBase-root-MuiIconButton-root" tabindex="0" type="button"><svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root" focusable="false" aria-hidden="true" viewBox="0 0 24 24" data-testid="GitHubIcon" style="width: 35px; height: 35px;">
        <path d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"></path>
      </svg></button><button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1fa0tqm-MuiButtonBase-root-MuiIconButton-root" tabindex="0" type="button"><svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root" focusable="false" aria-hidden="true" viewBox="0 0 24 24" data-testid="LanguageIcon" style="width: 35px; height: 35px;">
        <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"></path>
      </svg></button></div>
</div>"
`;
